packaging/packager_win.bat

```diff
@@ -0,0 +1,42 @@
+@Echo off
+
+SET DISTRO="windows"
+SET DISTRO_VERSION="10"
+SET DISTRO_ROOT="%HOMEDRIVE%"
+
+REM AutoPy is not available for Python3.9 at present
+SET DISABLE_AUTOPY=1
+REM Tesseract doesn't have original installers for Windows
+SET DISABLE_OCR=1
+REM Drag/drop with the current default DC backend is not supported on Windows
+SET DISABLE_DRAG=1
+REM Requires X server and thus only available on Linux
+SET DISABLE_XDOTOOL=1
+REM Requires VNC server and thus only available on Linux
+SET DISABLE_VNCDOTOOl=1
+
+REM Main deps
+cd %DISTRO_ROOT%\
+REM Assuming a local python executable to avoid overloading the download web page
+START /WAIT python-3.9.0-amd64.exe /quiet InstallAllUsers=1 PrependPath=1 Include_test=0
+REM set a temporary path variable valid inside this session
+set PYTHONPATH="C:\Program Files\Python39"
+set PATH=%PYTHONPATH%;%PYTHONPATH%\Scripts;%PATH%
+REM a permanent path variable will be set by the executable once this batch exits
+
+REM GuiBot deps
+pip install --user --upgrade pip
+pip install -r %DISTRO_ROOT%\guibot\packaging\pip_requirements.txt
+
+REM GuiBot setup
+echo Copying GuiBot files
+cd %DISTRO_ROOT%\guibot\packaging
+python setup.py install
+cd ..
+xcopy misc %PYTHONPATH%\Lib\site-packages\guibot\misc /E /S /I /Q /V
+xcopy tests %PYTHONPATH%\Lib\site-packages\guibot\tests /E /S /I /Q /V
+cd ..
+
+REM Run all unit tests
+cd %PYTHONPATH%\Lib\site-packages\guibot\tests
+python -m unittest discover -v -s ../tests/

```

Comment:
Don't `python -m unittest discover -v -s .` or simply `python -m unittest discover -v`  work here? You seem to be at the "tests" directory already

Reply:
Unfortunately not because the tests are located in a superdirectory and the current location is in the tests folder.
I took this from our run_tests bash script:
```
set -e

readonly libpath="${LIBPATH:-../guibot}"
readonly coverage="${COVERAGE:-coverage3}"

$coverage run --source="$libpath" -m unittest discover -v -s ../tests/
# use -i to ignore errors from pythong cache files and other traced dependencies
$coverage report -m -i
```
I could maybe try some variations of the idea.

Reply:
I also tried this which seemed more readable:
```diff
 REM GuiBot setup
 echo Copying GuiBot files
 cd %DISTRO_ROOT%\guibot\packaging
 python setup.py install
 xcopy %DISTRO_ROOT%\guibotmisc %PYTHONPATH%\Lib\site-packages\guibot\misc /E /S /I /Q /V
 xcopy %DISTRO_ROOT%\guibottests %PYTHONPATH%\Lib\site-packages\guibot\tests /E /S /I /Q /V
 
 REM Run all unit tests
-cd %PYTHONPATH%\Lib\site-packages\guibot\tests
-python -m unittest discover -v -s ../tests/
+cd %PYTHONPATH%\Lib\site-packages\guibot
+python -m unittest discover -v -s tests/
```
but
```
[stdlog] 2022-01-28 11:21:35,350 client           L0864 INFO | Using c:\program files\python39\lib\site-packages
[stdlog] 2022-01-28 11:21:35,350 client           L0864 INFO | Finished processing dependencies for guibot==0.41.1
[stdlog] 2022-01-28 11:21:35,350 client           L0864 INFO | Datei guibotmisc nicht gefunden
[stdlog] 2022-01-28 11:21:35,350 client           L0864 INFO | 0 Datei(en) kopiert
[stdlog] 2022-01-28 11:21:35,350 client           L0864 INFO | Datei guibottests nicht gefunden
[stdlog] 2022-01-28 11:21:35,350 client           L0864 INFO | 0 Datei(en) kopiert
[stdlog] 2022-01-28 11:21:35,350 client           L0864 INFO | Das System kann den angegebenen Pfad nicht finden.
[stdlog] 2022-01-28 11:21:35,541 client           L0864 INFO | Traceback (most recent call last):
[stdlog] 2022-01-28 11:21:35,542 client           L0864 INFO |   File "C:\Program Files\Python39\lib\runpy.py", line 197, in _run_module_as_main
[stdlog] 2022-01-28 11:21:35,542 client           L0864 INFO |     return _run_code(code, main_globals, None,
[stdlog] 2022-01-28 11:21:35,542 client           L0864 INFO |   File "C:\Program Files\Python39\lib\runpy.py", line 87, in _run_code
[stdlog] 2022-01-28 11:21:35,542 client           L0864 INFO |     exec(code, run_globals)
[stdlog] 2022-01-28 11:21:35,543 client           L0864 INFO |   File "C:\Program Files\Python39\lib\unittest\__main__.py", line 18, in <module>
[stdlog] 2022-01-28 11:21:35,543 client           L0864 INFO |     main(module=None)
[stdlog] 2022-01-28 11:21:35,543 client           L0864 INFO |   File "C:\Program Files\Python39\lib\unittest\main.py", line 100, in __init__
[stdlog] 2022-01-28 11:21:35,543 client           L0864 INFO |     self.parseArgs(argv)
[stdlog] 2022-01-28 11:21:35,543 client           L0864 INFO |   File "C:\Program Files\Python39\lib\unittest\main.py", line 124, in parseArgs
[stdlog] 2022-01-28 11:21:35,543 client           L0864 INFO |     self._do_discovery(argv[2:])
[stdlog] 2022-01-28 11:21:35,543 client           L0864 INFO |   File "C:\Program Files\Python39\lib\unittest\main.py", line 244, in _do_discovery
[stdlog] 2022-01-28 11:21:35,544 client           L0864 INFO |     self.createTests(from_discovery=True, Loader=Loader)
[stdlog] 2022-01-28 11:21:35,544 client           L0864 INFO |   File "C:\Program Files\Python39\lib\unittest\main.py", line 154, in createTests
[stdlog] 2022-01-28 11:21:35,544 client           L0864 INFO |     self.test = loader.discover(self.start, self.pattern, self.top)
[stdlog] 2022-01-28 11:21:35,544 client           L0864 INFO |   File "C:\Program Files\Python39\lib\unittest\loader.py", line 346, in discover
[stdlog] 2022-01-28 11:21:35,544 client           L0864 INFO |     raise ImportError('Start directory is not importable: %r' % start_dir)
[stdlog] 2022-01-28 11:21:35,544 client           L0864 INFO | ImportError: Start directory is not importable: 'tests/'
```

Reply:
I think we might need to turn the simple test directory into a python package to make this work, hence the previous implementation.

Reply:
I see
Alright, it's fine this way